domain/estimators/naive_bayes.py

```diff
@@ -32,21 +32,21 @@ def train(self):
 
     def predict_pp(self, row, attr, values):
         nb_score = []
+        correlated_attributes = self._get_corr_attributes(attr)
         for val1 in values:
             val1_count = self._freq[attr][val1]
-            log_prob = math.log(float(val1_count)/float(self._n_tuples))
-            correlated_attributes = self._get_corr_attributes(attr)
-            total_log_prob = 0.0
+            log_prob = math.log(float(val1_count) / float(self._n_tuples))
             for at in correlated_attributes:
-                if at != attr:
-                    val2 = row[at]
-                    val2_count = self._freq[at][val2]
-                    val2_val1_count = 0.1
-                    if val1 in self._cooccur_freq[attr][at]:
-                        if val2 in self._cooccur_freq[attr][at][val1]:
-                            val2_val1_count = max(self._cooccur_freq[attr][at][val1][val2] - 1.0, 0.1)
-                    p = float(val2_val1_count)/float(val1_count)
-                    log_prob += math.log(p)
+                # Ignore same attribute, index, and tuple id.
+                if at == attr or at == 'index' or at == '_tid_':
```

Comment:
Do we know why `index` might be included in the list of correlated attributes? We might run into issues here if there is genuinely a column in the data called `index`.

Reply:
it hasn't... I just noticed we do this in multiple places. I will remove the column `index` for now.

Reply:
I noticed we are doing this somewhere else so I followed the same convention. But I will remove it.